### PR TITLE
fix: add apache 2 license🚧

### PR DESCRIPTION
**Pull request by BelfordZ**
# THIS PROPOSAL HAS NOTHING TO DO WITH DCO, REAL NAMES, OR ANY RESTRICTIONS ON WHO CAN CONTRIBUTE

ref https://github.com/ethereumclassic/ECIPs/pull/162

All ECIP contributors who specified licenses other than apache 2.0 must resubmit their ECIP, removing the license. ( good job everyone, there are no examples of this.)

### CODE REVIEW

File: _specs/ecip-1000.md [hunk 1]
```diff
@@ -291,66 +284,7 @@ To avoid doubt: comments and status are unrelated metrics to judge an ECIP, and
...
-* GPL-2.0+: [GNU General Public License (GPL), version 2 or newer](http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html)
-* LGPL-2.1+: [GNU Lesser General Public License (LGPL), version 2.1 or newer](http://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html)
-
-## Rationale
-
-**Why are there software licenses included?**
-
-* Some ECIPs, especially consensus layer, may include literal code in the ECIP itself which may not be available under the exact license terms of the ECIP.
-* Despite this, not all software licenses would be acceptable for content included in ECIPs.
+You may not specify a license. Claims to any license other than the one specified in the repository root are invalid.
```
Comment (BelfordZ) [inline _specs/ecip-1000.md]:
```suggestion
You may not specify a license. Claims to any license other than the one specified [here](https://github.com/ethereumclassic/ECIPs/blob/master/LICENSE) are invalid.
```

**Comment by bobsummerwill** [comment]
We will also need DCOs from all existing contributors and confirmation that they agree to the relicensing.  The copyright sits with the contributors, not with the repository.   We need everyone's consent to relicense, but I think this a GREAT approach.   Let me stew on it and come back with feedback.   Will also raise with the IP lawyers for their input.

**Comment by TokenHash** [comment]
I disagree on changing licensing conditions in the current ECIP-1000 in this repo and making past ECIP contributors to agree on a single option retroactively based on no rational argument whatsoever.

Related to #162 and what is written in the comments here, there is no rational argument, even less rough consensus, on making past contributors retroactively agree on DCOs.

Any new changes conditions should apply going forward.

The option of "decloaking" retroactively or imposing using developer legal name should not even be  under discussion in ETC.

**Comment by bobsummerwill** [comment]
Just to be clear - we CANNOT commit this change (which relicenses the repository) without the consent of all of the contributors to date.  There are 55 of them.

**Comment by realcodywburns** [comment]
Ok. This is your wind mill to tilt at. I'm fine with mine being relicensed
 I have 5, 2 are cc0  there is never a situation where I will ever enforce copyright on any of my work.. This looks good to me as it is grammatical correct and functional accomplishes its stated goal. Please add a tracker to indicate progress
Adding wip to block merge

**Comment by bobsummerwill** [comment]
@realcodywburns Tracker:

https://docs.google.com/spreadsheets/d/1WxDIyg4s9O3FsrloJPt91c-2RWCgSUdEkpBaPpcw4E0/edit?usp=sharing

Looks like there are only 25 people to chase.
Shouldn't take long.
Mainly just an issue of consent.
How many anons contributing issues?  Just Dexeran.

Plus MikO seems to have done some README edits, and I don't know who "whilei" is but he/she seems to be at ETC Labs.

**Comment by TokenHash** [comment]
This PR is inappropriate, it solves no problem at all, restrictive, and it is authoritarian to create a root license, demand retroactively to previous contributors, and force future contributors to a root license with no justification whatsoever. 

As realcodyburns mentions above, It is also creating an unnecessary administrative burden, which is also authoritarian as it has no rationale, and disruptive of the ETC proposal process. The portfolio of options in ECIP-1000 is adequate as it is, and gives all contributors, with real names or anonymous, flexibility to work on the options they prefer.

ETC system does not have to be modified according to irrational rules with no fundament, which are restrictive, and authoritarian if no real problem is being solved.

I oppose to change root licensing of ECIPs, I oppose to make those changes retroactive, I oppose to make past contributors change their licenses. 

I also oppose requiring using real names to past and future contributors. I agree to discuss DCO and how to implement it going forward, but not retroactively.

In general, I oppose changes with no rationale and with no clear problem being solved, as they are arbitrary and open the door to future arbitrary and authoritarian behavior and change in ETC.

**Comment by phyro** [comment]
@TokenHash what do you mean by restrictive? as in you can't choose which license you want to apply to the ECIP? because it makes everything in the repo almost the most permissive you can get from what I understand.

Yes, the real name requirement seems like a really bad choice and doesn't really solve the underlying problem IMO.

**Comment by BelfordZ** [comment]
It's actually not re-licensing anything. Like I said, none of the ECIP in this repo actually ever did specify a license other than apache 2.0, and there are only a couple cases where anyone did specify apache 2.0 explicitly. 

image


Technically everything in this repo, unless otherwise stated, has been unlicensed. If there is any 'relicensing' happening here, its moving from no license -> a license. There is very good reasons to use apache 2.0 as the root license, and to discourage, and outright not allow contributions under anything else. 

Please give a thorough read through of this: https://opensource.google/docs/casebook/

I am also proposing that we apply apache 2.0, and never look back, never change the license, and never have any contributors sign any kind of CLA or additional agreement over and above what is laid out by apache 2.0. 

This is about technology, not politics. We should choose the license that is most permissive that absolves contributors of liability and warranty. 

to summarize:
1. Apache 2.0 
2. No CLA or DCO or ANYTHING outside of Apache 2.0
3. as the contributor, your free to make a PR removing your work if you dont like apache 2.0 for whatever reason.

**Comment by BelfordZ** [comment]
@TokenHash At the moment, it being unlicensed poses a risk to all contributors. This is not an attack, put away the pitch forks.

**Comment by TokenHash** [comment]
@BelfordZ 

`1. Apache 2.0
2. No CLA or DCO or ANYTHING outside of Apache 2.0
3. as the contributor, your free to make a PR removing your work if you dont like apache 2.0 for whatever reason.`

What is your proposal regarding requiring real names for devs.

**Comment by realcodywburns** [comment]
Several are cc0 licensed in the footer of the ecip

**Comment by TokenHash** [comment]
And the fact many ECIPs (the ones moved to "final" or "active" I guess are the important ones) don't have a license doesn't mean there has to be a blanket license and say goodbye to anyone who doesn't like it if there are other licenses that are ok as per ECIP-1000.

A better solution would be have a default license (say Apache 2+) but let devs to change that to another option in ECIP-1000 if they prefer so, rather than having them go away from ETC.

Can a root license be overridden by another license selected by the author of an ECIP?

**Comment by BelfordZ** [comment]
> What is your proposal regarding requiring real names for devs.

This is a seperate matter IMO. Let's keep it to one thing at a time.

> Several are cc0 licensed in the footer of the ecip

Ah i didnt see this, thanks for pointing it out. As it doesn't adhere to the specification laid out in ECIP-1000, I'd propose that its an invalid ECIP anyways, and thus should be removed and re-filled.

> And the fact many ECIPs (the ones moved to "Last Call", "final" or "active" I guess are the important ones) don't have a license doesn't mean there has to be a blanket license and say goodbye to anyone who doesn't like it if there are other licenses that are ok as per ECIP-1000....

Apache 2.0 being a permissive license allows for anyone to maintain a fork which is re-licensed under anything they want. Freedom of use is the goal here, right? I thought I could get by without having to actually argue for why Apache 2.0 is the most suitable license for this repository. There is a lot of writing on this matter, but generally takeaways are:

1. dont invent your own licenses
2. there is barely any legal precedent for most of OSS anyways, so its really just best guess here. see above-linked google's open source casebook
3. Copyleft is counterproductive in terms of collaboration with 99% of good developers: the ones with jobs.

Not including a license (like we have been) is great, but it also means there is no statement of liability or warranty. This is not good.

The concept of using a term such as "Apache 2.0+" is IMO an anti-pattern. If you didn't care about the version, why would you say 2.0 then.

I'm proposing that we stick to Apache 2.0, keep things simple, and stop yack shaving on this BS.

**Comment by sorpaas** [comment]
@BelfordZ 

> Not including a license (like we have been) is great, but it also means there is no statement of liability or warranty. This is not good.

I don't quite agree with that interpretation. Not including a license, in many jurisdictions, will mean it's by default "all rights reserved". That means it's forbidden to reproduce or modify.

**Comment by BelfordZ** [comment]
@sorpaas yeah, some say no license means 'any license', since a license can be added at any point without recourse.

**Comment by TokenHash** [comment]
Of 41 ECIPs (draft + last call + final + active):

• 9 use copyright/license CC0
• 2 use copyright/license Apache 2.0
• 30 have no license

Of the 9 that used CC0, they are 7 different authors who used it. One author used Apache 2.0 for 2 ECIPs.

I suspect the 30 which have no license it's because ETC was very disorganized before and the ECIP-X which is generally used as a template did not have a #Copyright/Licensing section by default so all authors omitted that detail.

I still don't see any rationale to establish a root license instead of the options given in ECIP-1000.

It is, from now on, according to ECIP-1000, a function of editors to actually supervise that authors include a copyright/license from the options.

I do agree to seek all past authors to add a license to their past ECIPs, I can do that. This is the doc with the stats: 

https://docs.google.com/spreadsheets/d/1mnpm8FC33dT4eKefDDMTIl2jPAKRrGvkQNPwDk3TUos/edit?usp=sharing

ECIPs and licensing

**Comment by zmitton** [comment]
I oppose the requirement of naming authors. I am in favor of attempting getting explicit licensing from past contribs and going forward.

An observation: Tracking down _all_ authors is not strictly necessary for proposals that are just a parameter choice like for instance ecip-1047 (but we already have the author for that one).

**Comment by bobsummerwill** [comment]
@BelfordZ "If there is any 'relicensing' happening here, its moving from no license -> a license."

Yes - we are saying the same thing, but with different words.
The licensing ambiguity is the worst element, because that leads to potential legal risk.

**Comment by bobsummerwill** [comment]
@TokenHash Thanks for the table.  That is awesome.

The reason that I propose Apache 2.0 is because it is the best of breed permissive license which adds patent protection in addition to clarity of copyright.

If we are contacting all prior contributors, it is really very little effort to ask if they would agree to relicensing their ECIP to Apache 2.0.   We need their consent to do so.  We cannot compel anybody to relicense.  They have a perfect right to say no.  But it does no harm to ask.

I suspect that just about everybody will not have a strong objection, because it makes no practical difference to them.  Most of them probably did not even think about licensing at the time.

If we end up with some "objectors" but can at least get to clarity of licensing for all ECIPs then that would be a good place to be as well.

Moving forward, though, I see little reason why we should NOT require Apache 2.0 for all submissions.

I think that it is also important to add tracking of this "manual consent" with DCOs, both backwards and forwards in time.   If there is strong objection to a real name requirement then let's just leave that alone and proceed with DCOs using the existing self-identifications which we already have.

Which is actually the same as "real names" for everyone except Dexeran!

**Comment by bobsummerwill** [comment]
BTW - Adding an Apache 2.0 license at the root would be a good thing.
That, plus a DCO bot, gives us "best of breed" moving forward, with minimal barrier of entry for new participants.  Just the need to do the "register your name and email address" the first time any of your changes are committed.

Anybody can create issues with no barrier.
Anybody can still comment on anything with no barrier.
Anybody can still create pull requests with no barrier.  They just cannot be merged until you register your name (which could be pseudonym) and email address - so you can be contacted in the future.

**Comment by bobsummerwill** [comment]
I would like a real name requirement too, because I suspect that DCOs with a pseudonym provide no legal protection - nullifying both the copyright and patent protections of the Apache 2.0 license and putting us in the same position as "public domain" for those contributions.

But maybe that makes no practical difference anyway?

It is only is a contribution from a bad actor gets all the way through to being accepted into the protocol that the damage can really happen.

We have never had any pseudonymous changes accepted into ETC.

The first ever pseudonymous change to be accepted into ETH would be ProgPOW.

**Comment by drd34d** [comment]
I oppose the requirement of author's real name. An alias and email should be sufficent for the purpose you're trying to accomplish. Requiring any contributor to give it's real name can be a barrier of entry and an oppressive thing to require that does not align well with the values of ETC.

**Comment by bobsummerwill** [comment]
@zmitton

"An observation: Tracking down all authors signatures is not strictly necessary for proposals that are just a parameter choice like for instance ecip-1047 (but we already have the author for that one)."

Correct.  A bunch of "contributors" to this repository were for EIP files in the original cloning and those files are since deleted.   Or contributions were typo fixes or readme changes or merging.  Small things which would need meet the requirement of "IP".   We can ignore those.

See https://docs.google.com/spreadsheets/d/14OlEk2MhsEr9vn5dqgVIw1h8mWqKymon3lbnwp1QwUY/edit#gid=871150240
where those are represented by crossed out green cells.

**Comment by TokenHash** [comment]
After reviewing all 127 BIPS, the Bitcoin BIP situation is similar to ECIP:

• Older BIPs (2011-2015) are not licensed or use `PD`
• In the "middle ages" (2015-2016) BIPs are mainly `PD`, which was later discouraged by BIP-0002 <- the equivalent of ECIP-1000 
• Newer BIPs (as from 2017) they all comply with BIP-0002 and all use formats such as `BSD-3-Clause | CC0-1.0` combined, just `BSD-2-Clause` or `CC0-1.0`

Summary: older BIPs are unlicensed or `PD`. Middle ages BIPs are `PD`. Everything in 2017 and after is licensed using best practices in BIP-0002.

**Comment by TokenHash** [comment]
To establish a blanket license, irrespective of which one and how good it is, and to exclude those who don't want to use it, when there are several variations in ECIP-1000 that are totally acceptable, as seen in ETC and Bitcoin, is a security hole as this "small" and what appears as an "irrelevant" action is an unnecessary restriction to a single license and establishes "restriction" in itself as a philosophical norm, in ETC.

There is a long tradition of "least authority" in crypto, but this is exactly the opposite: an unnecessary and not justified "most authority" action.

This, as a first action of establishing a "most authority" philosophy pushed by @bobsummerwill will inexorably be followed by promoting more restrictions as already mentioned by him:

• DCO
• naming developers and killing anonymity
• introducing more "protections" such as "trademarks and trade secrets"
• speaking with lawyers to appeal to authority to impose more restrictions
• appealing to EF, EEA, Hyperledger, and other "OSS foundations" as authority of how ETC should proceed, which are all opposite to ETC ethos

Summerwill, has the wrong mindset, and all other participants in ETC have a lazy mindset that prefer to move on with what they consider as "irrelevant" or "BS" or "a waste of time", but they don't see they are opening the door to further attacks with more restrictions in ETC in the future. Just because there is one attacker being bothersome enough to push all to an unnecessary change. 

The "I am protecting you" is the oldest excuse by authoritarians who then violate people with top down governance and restrictions, especially corporatist in this case. Summerwill has only spoken about top down authoritarian measures in ETC for months.

And "I am going to annoy you until you capitulate", and the fact all people in ETC actually capitulate and rationalize their capitulation, is another way of introducing these negative changes.

Just because ECIP-1000 was approved late in the life cycle of ETC (in 2019), that editors have not enforced the copyright/licensing section in new ECIPs, and that it is easy and comfortable to put a root license does not mean that can't be solved by using best practices from now on.

ETC is NOT Linux or Apache or any old OSS system. ETC is a system that needs to have the highest standards of trust minimization, unlike any other type of OSS system, to protect a global platform of money, property and agreements. This has nothing to do, or has anything in common, with other OSS systems, or private networks, that can be used in isolation or have no property and agreements inside, or are fine with top down trust maximization.

There is no reason to use a single root license and not do what Bitcoin already does ever since 2017 of using a portfolio of acceptable options. "I am lazy" and "this bother me", is not an argument. Therefore I oppose this action.

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (sorpaas) [inline LICENSE]: Please copy the full content of the license to this file. That will be much more clear and self-contained.

File: _specs/ecip-1000.md [hunk 3]
```diff
@@ -291,66 +284,7 @@ To avoid doubt: comments and status are unrelated metrics to judge an ECIP, and
...
-* GPL-2.0+: [GNU General Public License (GPL), version 2 or newer](http://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html)
-* LGPL-2.1+: [GNU Lesser General Public License (LGPL), version 2.1 or newer](http://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html)
-
-## Rationale
-
-**Why are there software licenses included?**
-
-* Some ECIPs, especially consensus layer, may include literal code in the ECIP itself which may not be available under the exact license terms of the ECIP.
-* Despite this, not all software licenses would be acceptable for content included in ECIPs.
+You may not specify a license. Claims to any license other than the one specified [here](https://github.com/ethereumclassic/ECIPs/blob/master/LICENSE) are invalid.
```
Comment (meowsbits) [inline _specs/ecip-1000.md]:
```suggestion
You may specify any license you want. However, in order for a proposal to be permanently accepted to this repository (eg. merged via a [Github Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)), it must be associated with an [Apache 2 License](https://www.apache.org/licenses/LICENSE-2.0.html). 

For authors with alternative license preferences, it is suggested that you choose an even more lenient license (eg. [The Unlicense](https://choosealicense.com/licenses/unlicense/)) so that your work can be copied verbatim to this repository by any enterprising member of "the community" artistic, enterprising, or selfless enough to lay jealous claim to your ideas. In the case that this path is undesirable, you are free to pursue licensing and distribution of your ideas at will in any forum and any manner you wish; please only be advised that it will probably be more profitable to collect payment for your ideas and work _before_ actually sharing them with a bunch of pseudo-anonymous crypto-anarchists you met on the internet.  

By leaving this section blank, when and if your document's content is officially and technically included in this repository's `master` branch (no matter the labeled _Status_ of the document), the document will, along with its fore-running peers, fall under the Apache 2 License prescribed. 
```

**Comment by meowsbits** [comment]
At least as far as United States copyright law pertains, this may be of interest:

> (b) In no case does copyright protection for an original work of authorship extend to any idea, procedure, process, system, method of operation, concept, principle, or discovery, regardless of the form in which it is described, explained, illustrated, or embodied in such work.
> https://www.copyright.gov/title17/92chap1.html#102

So while US authors would, without an explicit license, hold exclusive copy rights for their document itself, the ideas (and methods... etc.) communicated are (in the US) free as in free range.

**Comment by bobsummerwill** [comment]
And we still need to add some language about DCOs, and look at DCO bot options.

**Comment by meowsbits** [comment]
As far as the "real names" thing goes (and I can't seem to find the kernel for this thread.. ?) I don't understand why at all it's in the discussion here. Again, for the US at least, there is NO (as in NONE) relationship assumed or necessary between matters of copyright and legal identification. Ben Franklin wrote as @SilenceDogood extensively (among many other names as well), for example, and the legacy of copyright-happy royalty-collecting pseudonymous literaries extends so far and wide even @50-Cent wouldn't know where to start listing them.

**Comment by bobsummerwill** [comment]
@meowsbits On your US copyright point above, what I believe that is saying is that the copyright refers to the textual description itself, which authors have implicitly, whether or not they add "Copyright (c) 2019 Bob Summerwill" or not anywhere in the text.   You authored it, you have copyright.

If you take care to have PROOF of that (which, I think, in the old days, would mean sending a copy of your manuscript, and signing and dating it and keeping it in a safe, or getting it notarized or whatever - but which is trivial now for open source software by just working in the open) then it is very easy to enforce your copyright if somebody plagiarizes your work.   Hello, Justin Sun.

That copyright does not stop others from doing something equivalent, no.   That is where patents come in.  Which is why I care about using Apache 2.0.   Because I don't want a Craig Wright / nChain style patent trolling drama to happen to ETC because of the simple lack of IP protection which we are discussing here.

Something which still seems up in the air in the US at least is whether APIs are copyrightable, as horrible, horrible Oracle have been trying to do to Android by asserting that the Java APIs are their copyright, and even though Android is built on a completely "clean house" implementation of Java that they deserve billions of dollars:

https://en.wikipedia.org/wiki/Oracle_America,_Inc._v._Google,_Inc.

The current status is ... "Oracle appealed to the United States Court of Appeals for the Federal Circuit, which found in March 2018 that Google's reuse of the APIs had not been fair use, ruling in favor of Oracle."

It will likely go to the Supreme Court, where (hopefully) it will be struck down.  But it looks like this horror show is actually going to lead to Google ditching Android ultimately, for their own new OS with no Java dependencies.

This court case was a key reason why I strongly opposed any though (while we were looking at eWASM) of considering retargeted the EVM on top of the JVM.   Because of Oracle.  That is what Corda does, but that route is poison.    .NET VM is safer, but there is still a risk there, if Microsoft went rogue again.   Not sure of the IP situation on eWASM.

But the best of the lot there is sticking with EVM indefinitely, and iterating that along - total self-sovereignty.   Which is what @gcolvin always advocated.  Hurrah for Greg.

**Comment by bobsummerwill** [comment]
DEVCON3 panel - evolving the EVM - where Greg spoke on this:

https://www.youtube.com/watch?v=txBl6ekcZXY

**Comment by meowsbits** [comment]
Here's a nice little thing curated by Github on ["Here's what happens if you don't choose a license"](https://choosealicense.com/no-permission/).

**Comment by bobsummerwill** [comment]
@meowsbits The real name part is not to do with collecting royalties.  It is the other way around.

If somebody contributes a change through a pseudonym then the copyright and patent protection you think you are getting through using Apache 2.0 with DCOs is nullified.  Because somebody "signed the contract" with a fake name.   You might be in receipt of code/spec they copied from elsewhere and the original author can then make legal claims against ETC users/companies/exchanges etc.

You might have patented content which you "sucked up" which they slipped in through this pseudonymous approach.  There is nobody to go and kick in the balls.  At least with a real name requirement, if some patsy is put up as a front you can go and get them.

Ultimately all of this is just game theory.  There is no perfect answer, but having no defences is the very worst answer.   The better defences we can have (as long as they are not adding too many friction - which I don't think any of this will) the less likely we are to be attacked.

Maybe social defence is good enough for ETC?   Because we are so practiced at spotting social attacks.   Whatever the case, if "real name" is too much then it is too much, and that is fine.

Kernel of that thread was https://github.com/ethereumclassic/ECIPs/pull/162, which I withdrew because @BelfordZ's proposal here was much simpler and better.

**Comment by meowsbits** [comment]
> If somebody contributes a change through a pseudonym then the copyright and patent protection you think you are getting through using Apache 2.0 with DCOs is nullified.

Can you cite this?

**Comment by meowsbits** [comment]
And @bobsummerwill 

> That is where patents come in. Which is why I care about using Apache 2.0. [...] Because I don't want [...] patent trolling drama ... 

Can you explain with citations how using an Apache 2 License solves or avoids a patent troll scenario?

**Comment by bobsummerwill** [comment]
@sorpaas "Given that this has become a controversial issue, we might want to be careful merging it."

More than careful.   We SHOULD NOT merge it until we have resolution on all the outstanding issues - specifically getting consent from all prior contributors - because addition of a repo-level license is re-licensing - and the copyright lies with the original authors.   It was not "assigned".

Nevertheless, I think this is the best approach.  We are making rapid progress.

**Comment by bobsummerwill** [comment]
@meowsbits "Can you explain with citations how using an Apache 2 License avoids or would solve a patent troll scenario?"

Sure.  The Apache 2.0 license states that any contributors grant rights to use of any patents they hold on their contribution.  So if you hold patents then you cannot enforce them against the project.   You could enforce them against projects which are using MIT, BSD or public domain licensing though, because there was no such corresponding grant.

That is patent trolling in a nutshell.   Sticking your stuff into unsuspecting projects and then making $$$ by saying "I have a patent which means I have exclusive rights to monetize this thing for N years, so you owe me a million dollars".   That happened to Linux, and it has happened to many, many other projects with over-weak permissive licensing.   It is the reason the GPLv3 exists too.  The GPLv2 used for the Linux Kernel did not have that protection either, and so Microsoft had a multi-billion dollar per year business trolling their patents onto Android companies.  Over $1B per year just from Samsung to Microsoft.

- https://tldrlegal.com/license/mit-license
- https://tldrlegal.com/license/bsd-3-clause-license-(revised)
- https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)
- https://tldrlegal.com/license/gnu-general-public-license-v2
- https://tldrlegal.com/license/gnu-general-public-license-v3-(gpl-3)
- http://www.ifross.org/en/what-difference-between-gplv2-and-gplv3
- https://resources.whitesourcesoftware.com/blog-whitesource/top-10-apache-license-questions-answered

"Explicit grant of patent rights: Apache License 2.0 explicitly lays down the grant of patent rights while using, modifying or distributing Apache licensed software; it also lists the circumstances when such grant gets withdrawn."

**Comment by bobsummerwill** [comment]
BTW - Microsoft have done a complete 180 in the last few years under Satya Nadella, and are utterly unlike Steve Ballmer's "Linux is a Cancer" Microsoft.  There are real allies of open source now.

- https://arstechnica.com/tech-policy/2014/10/lawsuit-reveals-samsung-paid-microsoft-1-billion-a-year-for-android-patents/
- https://www.androidpolice.com/2018/10/10/report-microsoft-making-patents-royalty-free-android-oems/
- https://www.zdnet.com/article/microsoft-open-sources-its-entire-patent-portfolio/

See also https://www.hanselman.com/blog/MicrosoftKilledMyPappy.aspx.

This is NOT to say that Microsoft is not what we would like it to be in other axis, but for open source they are very good now.  Windows has "spyware" in it.   They are legally compliant so will shut down things on GitHub when asked to by governments.

So they are not aligned on censorship resistance, but purely on the "open source friendly", we don't need to be afraid of them anymore.  Apple are "the baddies" on that score, but many people give them a pass because of their privacy-respecting best practices.   They do that because privacy is an absolute competitive advance for Apple over Google - not because they are "good people":

https://ar.al/notes/apple-vs-google-on-privacy-a-tale-of-absolute-competitive-advantage/

File: LICENSE (hunk 2 above)
Comment (BelfordZ) [reply LICENSE]: This is the suggested format, as per apache 2.0 instructions. I was on the fence about it, but decided following their mandate is the way to go. 

Please consider this. If you still disagree, I will change it.

Comment (YazzyYaz) [reply LICENSE]: For your reference: https://www.apache.org/licenses/LICENSE-2.0

File: _specs/ecip-1000.md (hunk 3 above)
Comment (BelfordZ) [reply _specs/ecip-1000.md]: I disagree. It presents unnecessary challenges and risks to allow people to license the contents of this repo differently. 1 GPL ECIP and the whole repo is tainted.

**Comment by BelfordZ** [comment]
Im going to start deleting any comments or suggestions relating to DCO, or real names. I hope you understand why censorship is required here.

**Comment by bobsummerwill** [comment]
"Im going to start deleting any comments or suggestions relating to DCO, or real names. I hope you understand why censorship is required here."

It is *not* required, @BelfordZ.  Why censor?  Just ignore anything you don't agree with.

**Comment by BelfordZ** [comment]
Because the derailing of conversation is wasting mine, and everyones time.

**Comment by TokenHash** [comment]
Hi @BelfordZ to answer only about the topic, this PR does two things

(1) Proposes Apache 2.0
(2) Restricts the ECIP to to only Apache 2.0

I am in favor of (1) Apache 2.0 as a license to be used in the ECIP process, I am not in favor of (2) establishing only one license as the only option in the ECIP process.

I prefer to stay with the options given in ECIP-1000 which are:

Recommended licenses:

Apache-2.0: Apache License, version 2.0
BSD-2-Clause: OSI-approved BSD 2-clause license
BSD-3-Clause: OSI-approved BSD 3-clause license
CC0-1.0: Creative Commons CC0 1.0 Universal
GNU-All-Permissive: GNU All-Permissive License
In addition, it is recommended that literal code included in the ECIP be dual-licensed under the same license terms as the project it modifies. For example, literal code intended for Ethereum Classic Core would ideally be dual-licensed under the MIT license terms as well as one of the above with the rest of the ECIP text.

Not recommended, but acceptable licenses:

BSL-1.0: Boost Software License, version 1.0
CC-BY-4.0: Creative Commons Attribution 4.0 International
CC-BY-SA-4.0: Creative Commons Attribution-ShareAlike 4.0 International
MIT: Expat/MIT/X11 license
AGPL-3.0+: GNU Affero General Public License (AGPL), version 3 or newer
FDL-1.3: GNU Free Documentation License, version 1.3
GPL-2.0+: GNU General Public License (GPL), version 2 or newer
LGPL-2.1+: GNU Lesser General Public License (LGPL), version 2.1 or newer

Rationale:

The philosophy behind the above is that it is preferable for ETC to follow a "least authority" principle only establishing rules that are no more than is absolutely necessary. Although Apache 2.0 is fine, it is not absolutely necessary to limit the whole repo to only that license. Therefore I prefer to continue with the portfolio of options that exists in ECIP-1000.

Logistics:

As to logistics going forward, I think we can very easily align all previous contributions to one of the licenses in ECIP-1000, I can help with that, and from now on the ECIP editors can perfectly bounce any ECIP that doesn't have a correct license. So I think  will be in order from now on.

Multiple unrelated topics:

As this PR is one step in a series of "protections" to ETC I thought it was relevant to include the other issues as related externalities. However, I agree with you that this thread relates to only the above issue, so I will stick to that rule from now on.

**Comment by BelfordZ** [comment]
Thank you @TokenHash for the well constructed criticism. 

Here are the main reasons for why I propose apache 2.0 for all, without exception:
1. Avoid copyleftist licenses and all licenses which have political implications
1. Avoid contagious licenses that may taint the validity of Apache 2.0. 
1. Avoid unknown issues that may arise in having a project that contains multiple licenses.
1. Any works licensed under Apache 2.0 can be copied and relicensed without issue - If you want slightly chance an ECIP and relicense it, patent it, what ever, go ahead. Apache 2 is freedom. Everything else is politically infused horse shit, or provides unknown legal protections.

That said, this isnt a hill im willing to die on. Should this be a deal breaker for you folks, I'm happy to meet halfway, adding root license of apache 2.0, allowing other licenses to be specified, but I would urge that any GNU, GPL, or as I put, politically infused licenses not be included in the allowable list.

**Comment by TokenHash** [comment]
Thanks @BelfordZ,

> • Avoid copyleftist licenses and all licenses which have political implications
> • Avoid contagious licenses that may taint the validity of Apache 2.0.
> • Avoid unknown issues that may arise in having a project that contains multiple licenses.
> • Any works licensed under Apache 2.0 can be copied and relicensed without issue.

Those are reasonable arguments. Especially the possible mixing of licenses that may invalidate or neutralize the others, thus possibly defaulting to copyright or other nefarious stuff.

**Comment by TokenHash** [comment]
@BelfordZ 

>Should this be a deal breaker for you folks, I'm happy to meet halfway, adding root license of apache 2.0, allowing other licenses to be specified, but I would urge that any GNU, GPL, or as I put, politically infused licenses not be included in the allowable list.

Yes, let's meet in the middle. Allowing Apache 2.0 as the root license so ETC has a default option, but having a more extended list in ECIP-1000 that allows devs to use alternatives, but perhaps post them in another repo or site e.g. ecips.that.world as @sorpaas has done in the past.

The ETC ecosystem can use ECIP-1000 in this Github Organization (https://github.com/ethereumclassic/ECIPs) as the canonical reference of which licenses to use.

@BelfordZ can you point out which licenses in ECIP-1000 are hyper-toxic to you?

I just speak for myself here, others may have another opinion.

**Comment by BelfordZ** [comment]
yeah im over it

**Comment by bobsummerwill** [comment]
"Allowing Apache 2.0 as the root license so ETC has a default option, but having a more extended list in ECIP-1000 that allows devs to use alternatives, but perhaps post them in another repo or site e.g. ecips.that.world as @sorpaas has done in the past."

As long as those other repos are suitably constrained to have to use one of a very short list of acceptable alternatives, which I think likely boils down to only Apache 2.0 or the new Chinese license.   So maybe we end up with another repo for proposals using that license, like @kimisan has said he would prefer?   But we really don't want loads of repos, because it makes it horrible for people to work out where everything is.

Also - even just to add the Apache 2.0 license in the root here, we still need to get consent from the previous contributors - with my spreadsheet and Donald's getting close to that already.

File: _specs/ecip-1000.md (hunk 3 above)
Comment (meowsbits) [reply _specs/ecip-1000.md]: ?

> You may specify any license you want. However, in order for the document content of a proposal to be permanently accepted to this repository (eg. merged via a [Github Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests)), it must be associated with an [Apache 2 License](https://www.apache.org/licenses/LICENSE-2.0.html).
